src/centered-on-asteroid/train.py

```diff
@@ -141,27 +148,227 @@ def train(
+    )
+    parser.add_argument(
+        "--experiment_name",
+        type=str,
+        default="end-to-end",
+        help="Name of the experiment (This will be combined with a timestamp to create a unique identifier). Default is 'end-to-end'.",
+    )
 
-    model = model_def.CFN(images_per_sequence, feature_vector_size, image_shape)
+    # Data
+    parser.add_argument(
+        "--metadata",
+        type=str,
+        default="no_metadata",
+        help="Metadata to use. Default is 'no_metadata'. Choices are: 'no_metadata', 'max_grad_diff', 'max_ang_diff', 'max_movement_vector_distance', 'max_movement_vector_distance_normalised', 'gradients', 'angles', 'movement_vectors'.",
+    )
+    parser.add_argument(
+        "--data_path",
+        type=str,
+        default="./../../scripts/data",
+        help="Path to the data folder.",
+    )
+
+    parser.add_argument(
+        "--images_folder",
+        type=str,
+        default="/images/centered_on_asteroid",
+        help="Path to the images folder. This will use the data_path as the root. Default is '/images/centered_on_asteroid'.",
+    )
+
+    parser.add_argument(
+        "--real_movers_file",
+        type=str,
+        default="/csv/movers_cond_12_image_meta_data.csv",
+        help="Path to the real movers file. This will use the data_path as the root. Default is '/csv/movers_cond_12_image_meta_data.csv'.",
+    )
+
+    parser.add_argument(
+        "--bogus_movers_file",
+        type=str,
+        default="/csv/movers_cond_2_image_meta_data.csv",
+        help="Path to the bogus movers file. This will use the data_path as the root. Default is '/csv/movers_cond_2_image_meta_data.csv'.",
+    )
+
+    # Add in future file system (where the movers are all stored in one file)
+    parser.add_argument(
+        "--movers_file",
+        type=str,
+        default="/csv/movers.csv",
+        help="Path to the (all) movers file. This will use the data_path as the root. Default is '/csv/movers.csv'.",
+    )
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    # Loss lookup table
+    classification_loss_functions = {
+        "CrossEntropyLoss": nn.CrossEntropyLoss(),
+        "NLLLoss": nn.NLLLoss(),
+        "BCELoss": nn.BCELoss(),
+        "BCEWithLogitsLoss": nn.BCEWithLogitsLoss(),
+        "MultiLabelSoftMarginLoss": nn.MultiLabelSoftMarginLoss(),
+        "MultiMarginLoss": nn.MultiMarginLoss(),
+        "SoftMarginLoss": nn.SoftMarginLoss(),
+    }
+    args = get_experiment_args()
+    # Model parameters
+    image_shape = args.image_shape
+    images_per_sequence = args.images_per_sequence
+    feature_vector_size = args.feature_vector_size
+
+    # CNN parameters
+    num_conv_blocks = args.num_conv_blocks
+    conv_filters_list = list(map(int, args.conv_filters_list.split(",")))
+    conv_kernel_sizes = args.conv_kernel_sizes
+    conv_strides = args.conv_strides
+    conv_padding = args.conv_padding
```

Comment:
Maybe pass all of them directly into the function instead of creating a new variable here